### PR TITLE
Add reference scripts

### PR DESCRIPTION
**Pull request by WhatisRT**
Adds reference scripts to the Babbage spec

### CODE REVIEW

File: eras/babbage/formal-spec/utxo.tex [hunk 1]
```diff
@@ -5,6 +5,13 @@ \section{UTxO}
   \emph{Functions}
   %
   \begin{align*}
+    & \fun{isTwoPhaseScriptAddress} : \Tx \to \UTxO \to \Addr \to \Bool \\
+    & \fun{isTwoPhaseScriptAddress}~tx~utxo~a = \\
+    &\quad\begin{cases}
+        \True  & a \in \AddrScr \land \fun{validatorHash}~a \mapsto s \in \fun{txscripts}~tx~utxo \land s \in \ScriptPhTwo \\
```
Comment (JaredCorduan) [inline eras/babbage/formal-spec/utxo.tex]: maybe highlight the new `utxo` parameter so we can see what changed?

File: eras/babbage/formal-spec/utxo.tex [hunk 2]
```diff
@@ -46,6 +53,13 @@ \section{UTxO}
         [\var{d}] & \hldiff{(\_, \_, d) \in \var{utxo}~\var{sp},~ \var{d} \in \Datum} \\
         \epsilon  & \text{otherwise}
       \end{cases}
+    \nextdef
+    & \fun{txscripts} : \Tx \to \UTxO \to \powerset{\Script} \\
+    & \fun{txscripts}~tx~utxo = \fun{txwitscripts}~tx \cup \{ \fun{hash}~s \mapsto s \mid (\_, \_, \_, s) \in \var{utxo}~(\fun{spendInputs}~tx \cup \fun{refInputs}~tx)\}
```
Comment (JaredCorduan) [inline eras/babbage/formal-spec/utxo.tex]: maybe highlight from the union onward so we can see what changed?

File: eras/babbage/formal-spec/utxo.tex [hunk 3]
```diff
@@ -46,6 +53,13 @@ \section{UTxO}
...
+    \nextdef
+    & \fun{languages} : \Tx \to \UTxO \to \powerset{\Language} \\
+    & \fun{languages}~tx~utxo =
+      \{\fun{language}~s \mid s \in \range (\fun{txscripts}~tx~utxo) \cap \ScriptPhTwo\}
```
Comment (JaredCorduan) [inline eras/babbage/formal-spec/utxo.tex:62]: same as earlier, maybe highlight the new utxo parameter so we can see what changed?

**Review by JaredCorduan — APPROVED** [review]
I made some suggestions about highlighting, but other than that looks great

File: eras/babbage/formal-spec/transactions.tex [hunk 4]
```diff
@@ -14,8 +14,8 @@ \section{Transactions}
   \begin{equation*}
     \begin{array}{r@{~~}l@{~~}l@{\qquad}l}
       \TxBody ~=~
-      & \powerset{\TxIn} & \fun{txinputs}& \text{Inputs}\\
-      & \times ~\powerset{\TxIn} & \fun{collateral} & \text{Collateral inputs}\\
+      & \powerset{\TxIn} & \fun{spendInputs}& \text{Inputs}\\
+      & \times ~\powerset{\TxIn} & \fun{collInputs} & \text{Collateral inputs}\\
```
Comment (polinavino) [inline eras/babbage/formal-spec/transactions.tex:18]: I think you need to rename the functions at use sites as well

**Pull request merged**